### PR TITLE
Add sengled water leak sensor device

### PR DESCRIPTION
**Pull request by anzhuge**
1. Modify fingerprints.yml file for sengled water leak sensor.
2. Add the sendled directory under the src directory and add init.lua file under sendled directory.

### CODE REVIEW

**Comment by lelandblue** [comment]
Hello - @anzhuge the init.lua file in ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled is currently uploaded as a submodule. Please update this PR so that the init.lua file is present without a new submodule.

Also, please update the PR name to properly describe this PR, thanks!

you can refer to this other sengled device PR for reference: https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/374

**Comment by anzhuge** [comment]
Hello ***@***.***

As per your request, I have made new modifications and resubmitted the PR at: https://github.com/anzhuge/SmartThingsEdgeDrivers/pull/1




comments:
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.


Please review and contact us promptly if you have any questions.










At 2023-04-20 00:09:39, "lelandblue" ***@***.***> wrote:

Hello - @anzhuge the init.lua file in ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled is currently uploaded as a submodule. Please update this PR so that the init.lua file is present without a new submodule.

Also, please update the PR name to properly describe this PR, thanks!

you can refer to this other sengled device PR for reference: #374

**Comment by anzhuge** [comment]
I'm sorry, please ignore the content.
















At 2023-04-23 19:30:40, "朱葛" ***@***.***> wrote:

Hello ***@***.***

As per your request, I have made new modifications and resubmitted the PR at: https://github.com/anzhuge/SmartThingsEdgeDrivers/pull/1




comments:
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.


Please review and contact us promptly if you have any questions.










At 2023-04-20 00:09:39, "lelandblue" ***@***.***> wrote:

Hello - @anzhuge the init.lua file in ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled is currently uploaded as a submodule. Please update this PR so that the init.lua file is present without a new submodule.

Also, please update the PR name to properly describe this PR, thanks!

you can refer to this other sengled device PR for reference: #374

**Comment by anzhuge** [comment]
Hello ***@***.***

As per your request, I have made new modifications and committed.




comments:
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.


Please review and contact us promptly if you have any questions.













在 2023-04-23 19:50:19，"朱葛" ***@***.***> 写道：

I'm sorry, please ignore the content.
















At 2023-04-23 19:30:40, "朱葛" ***@***.***> wrote:

Hello ***@***.***

As per your request, I have made new modifications and resubmitted the PR at: https://github.com/anzhuge/SmartThingsEdgeDrivers/pull/1




comments:
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.


Please review and contact us promptly if you have any questions.










At 2023-04-20 00:09:39, "lelandblue" ***@***.***> wrote:

Hello - @anzhuge the init.lua file in ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled is currently uploaded as a submodule. Please update this PR so that the init.lua file is present without a new submodule.

Also, please update the PR name to properly describe this PR, thanks!

you can refer to this other sengled device PR for reference: #374

**Comment by anzhuge** [comment]
1. add the init.lua file to the path ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\src\sengled.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-water-leak-sensor\fingerprints.yml file, a new device type E1L-G7K is added.

**Comment by anzhuge** [comment]
Hello, @lelandblue,

File: drivers/SmartThings/zigbee-water-leak-sensor/src/sengled/init.lua [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  battery_defaults.build_linear_voltage_init(2.6, 3.0)(driver, device)
+
+  for _, attribute in ipairs(CONFIGURATIONS) do
+    device:add_configured_attribute(attribute)
+    device:add_monitored_attribute(attribute)
+  end
+end
+
+local sengled_water_leak_sensor_handler = {
+  NAME = "Sengled Water Leak Sensor  Handler",
```
Comment (ctowns) [inline drivers/SmartThings/zigbee-water-leak-sensor/src/sengled/init.lua:49]: nit:
```suggestion
  NAME = "Sengled Water Leak Sensor Handler",
```

File: drivers/SmartThings/zigbee-water-leak-sensor/fingerprints.yml [hunk 2]
```diff
@@ -99,3 +99,8 @@ zigbeeManufacturer:
     manufacturer: THIRDREALITY
     model: 3RWS18BZ
     deviceProfileName: water-battery
+  - id: "Sengled E1L-G7K"
+    deviceLabel: Sengled Water Leak Sensor
+    manufacturer: sengled.
```
Comment (ctowns) [inline drivers/SmartThings/zigbee-water-leak-sensor/fingerprints.yml:104]: I believe the manufacturer name should just be "sengled"
```suggestion
    manufacturer: sengled
```

**Review by ctowns — COMMENTED** [review]
Also, please squash your commits before merging

**Comment by anzhuge** [comment]
Add Sengled rgbw bulb device
Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-switch\src\rgbw-bulb\init.lua file, a new device type E23-N13 is added.
Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-switch\fingerprints.yml file, a new device type E23-N13 is added.

**Comment by anzhuge** [comment]
Add Sengled dimming light bulb device:
1. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-switch\src\zigbee-dimming-light\init.lua file, a new device type E23-N11 is added.
2. Modified ~\SmartThingsEdgeDrivers\drivers\SmartThings\zigbee-switch\fingerprints.yml file, a new device type E23-N11 is added.

**Comment by lelandblue** [comment]
@ctowns / @greens Can you please review this?

**Review by greens — CHANGES_REQUESTED** [review]
sub-drivers must be `require`d in the base driver in order to be loaded and receive commands

**Comment by greens** [comment]
conflicts need to be addressed with a rebase of the fork or the branch to origin/main

**Comment by anzhuge** [comment]
> conflicts need to be addressed with a rebase of the fork or the branch to origin/main

I updated a color light and a white light on PR 615#. Can this be allowed?
PR 615# is my previous PR update on water leak sensor device.

**Pull request closed without merging**